OneCommunityGlobal/HGNRest
# Xiaow cleanup old timer and modify timer behavior for better UX

# Description
This PR cleans up the old timer related code and files, and modify timer behavior. Now if users stop timer before it runs out and log the passed time, the remaining time will be come the goal for the next timer, and if timer run out by itself and user log the time, the goal for next timer will be set back to the initial goal.

## Related PRS (if any):
This backend PR is related to the frontend [PR#1455](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1455#issue-1962550930).

## Main changes explained:
- Delete old timer related files
- Update new timer clientHandler file
- add one field called `initialGoal` in the timer model
…

## How to test:
1. check into current branch, check into frontend [PR#1455](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1455#issue-1962550930)
2. do `npm install` and `npm run dev` to run this PR locally
3. log as any user
4. go to dashboard
5. verify that 
  5.1 when logging time before timer runs out, the next goal is what is remaining from last timer
  5.2 when logging time after timer runs out, the next goal is what is set as the initial goal.
  5.3 try to break it.
  **5.4 (new) now if there are multiple tabs (on the same browser or different browsers), once timer runs out, the ringing effect should be consistent if user add more or log time on any one tab.**

## Screenshots or videos of changes:
how to set the initial goal (used to be Goal for today):
![image](https://github.com/OneCommunityGlobal/HGNRest/assets/77769556/02f63a19-eab6-497e-901d-9c26c1296dd2)


## Note:
The initial goal can only be set  by editing and saving goal at the bottom of the timer modal.
